doc/data-protocol.md

```diff
@@ -0,0 +1,154 @@
+
+The Flight Controller on the Ground Station 'Gateway' is a preliminary solution to have the Control computer talk to the Rocket's Flight Controller during the first static fire
+
+# Development plan
+
+The work on this data protocol is still preliminary. The development will be iterative and the first milestone is the static fire planned for the end of the summer 2020. The protocol will be refined later on to better allocate the telemetry sent back to the Ground Station.
+
+## Static Fire
+
+Drivers:
+
+- Send commands to the Rocket and have the relevant ones forwarded to the Engine Computer
+- Send back Telemetry to the Ground Station, mainly comprised of measurement samples from the Engine Computer
+- No logic on the Flight Controller for the Telemetry packets, the packets are simply forwarded to the Ground Station
+
+## Launch
+
+Drivers:
+
+- Maximize the amount of data downlinked from the Rocket, by optimizing how the data packets are processed by the Flight Controller and maybe also the Ground Station
+
+# Serial links
+
+All serial links share the same configuration. The baud rate is `57600`.
+
+A frame on the serial links looks like this:
+
+| Byte number | Description             |
+| ----------- | ----------------------- |
+| 0           | Frame separator: `0x0A` |
+| 1           | Frame separator: `0x0D` |
+| 2           | Frame ID                |
+| 3           | Data byte 0             |
+| 4           | Data byte 1             |
+| [...]       | [...]                   |
+| N+3         | Data byte N             |
+
+The length of the data field is variable
+
+# CAN Bus
+
+> **Note:** the CAN bus between the Engine Computer and the Flight Controller can be extended to additional boards in future developments
+
+CAN bus: CAN2.0 (FlexCAN) @1Mbit/s with termination 120$\Omega$ resistors on the Flight Controller side and on the Engine Computer side
+
+Base frame format (11 identifier bits)
+
+# Data protocol
+
+The term `ID` is used below to denote `identifier field` of the CAN Bus and byte 2 (`Frame ID`) of a frame over the serial links
+
+The `identifier field` is an 11 bits field but only the 8 lower bits are used to make the data on the CAN Bus as similar as possible to the data on the serial buses.
+
+Each message on the Telecommand link and the Telemetry link has a unique ID
+
+| ID              | Frame type | From              | To                |
+|-----------------|------------|-------------------|-------------------|
+| `0x00` - `0x0F` | TC         | Ground            | Engine Computer   |
+| `0x1F` - `0x7F` | TM         | Engine Computer   | Ground            |
+| `0x80` - `0x8F` | TC         | Ground            | Flight Controller |
+| `0x90` - `0xEF` | TM         | Flight Controller | Ground            |
+| `0xF0` - `0xFF` | N/A        | N/A               | N/A               |
+
+The `ID` values `0xF0` - `0xFF` are left unused and will be used if new devices are added to the on-board CAN Bus
```

Comment:
@eweilow do you think this is enough for the needs of the _Engine Computer_ ?

The separation between what is TC and what is TM is flexible but the separation between the frames for the _Engine Computer_ and the frames for the _Flight Controller_ are likely to remain the same until launch once this PR is approved

Reply:
Am I reading it correctly in that there'll be **16** (0x0F - 0x00 = 16) different message types available to send to the engine computer and **96** (0x7F - 0x1F = 96) message types available to send back to the ground from the engine computer?

That should be more than plenty 👍

Reply:
Exactly !

Reply:
Hum.. I just proof read and there is a gap between the _Engine Computer_ TC IDs and the _Engine Computer_ TM IDs.

TM should be `0x10` - `0x7F` so it's **112** message IDs in total for the _Engine Computer_ Telemetry (and **95** message IDs for the _Flight Controller_)

Reply:
I'll fix that